### ISSUE TITLE
terminate called after throwing an instance of 'std::runtime_error'

### ISSUE THREAD

**Issue by chschroeder** [facebookresearch/fastText#465]
This occurred using Python 3.6 / Debian Jessie.

Scenario:
I am trying to optimize the hyperparameters. In this setting an unfortunate combination of parameters results in a very high loss, ultimately produces NaN values and terminates my whole program.

I am using fasttext via the python wrapper. It seems i cannot handle this from the python side.

````
[Read 4M words
Number of words:  30836
Number of labels: 0
Progress:   0.9% words/sec/thread:   21890 lr:  0.495586 loss: 20.253834 ETA:   Progress:   1.7% words/sec/thread:   38377 lr:  0.491272 loss: 21.234467 ETA:   0h 0mterminate called after throwing an instance of 'std::runtime_error'
terminate called after throwing an instance of 'std::runtime_errorterminate called after throwing an instance of 'terminate called after throwing an instance of 'terminate called after throwing an instance of ''
terminate called after throwing an instance of 'std::runtime_error'
std::runtime_error'
terminate called after throwing an instance of 'std::runtime_error'
  what():  Encountered NaN.
std::runtime_error'
std::runtime_error  what():  '
terminate called after throwing an instance of 'std::runtime_error'
Aborted]
````

**Comment by ariwaranosai** [comment]
This error should be translated into RuntimeError by pybind11.
You can try to catch RuntimeError. For me, this works.

**Comment by chschroeder** [comment]
Thanks for your advice. Unfortunately this does not seem to work for me.
I figured out catching it seems to work for `train_supervised`, but not for `train_unsupervised`.
I am using the latter currently.

**Comment by blodemaster** [comment]
@chschroeder It says "Number of labels: 0". That appears to be something that could mess things up. What are you trying to accomplish?

**Comment by eduamf** [comment]
> Thanks for your help. Unfortunately this does not seem to work for me.
> I figured out that catching the error seems to work for `train_supervised`, but not for `train_unsupervised`.
> I am using the latter currently.

No, I`m using to create word`s representation on unsupervised mode. 

fasttext cbow -input data/pre_homonimos.txt \
              -output models/cbowN \
              -lr 0.5 -epoch 160160 -dim 12 -minCount 2 -minn 2 -maxn 5 -ws 10 -loss hs -thread 3 

```
terminate called after throwing an instance of 'std::runtime_error'
terminate called recursively
  what():  Encountered NaN.
terminate called recursively
```

After reduce the number of epochs the error disappear. Using the same crazy number 160160, but with dimension like 3, the error disappear too. Probably, it`s an memory error.
